stdlib/_typeshed/__init__.pyi

```diff
@@ -53,7 +53,7 @@ class SupportsRDivMod(Protocol[_T_contra, _T_co]):
 
 class SupportsLenAndGetItem(Protocol[_T_co]):
     def __len__(self) -> int: ...
-    def __getitem__(self, __k: int) -> _T_co: ...
+    def __getitem__(self, __k: SupportsIndex) -> _T_co: ...
```

Comment:
I don't think we should change this, unless consumers of this protocol are actually passing non-int `__index__`ed objects to it. There's a good chance users passing a custom object will just use int instead of SupportsIndex as their annotation, and that likely will work just fine at runtime.

Reply:
Makes sense — should I revert the changes to `typing.Sequence` and `typing.MutableSequence` as well, do you think?

Reply:
The argument there isn't as strong because those aren't Protocols, so people have to subclass them explicitly and will get a nice error from mypy telling them what's wrong (see mypy-primer output).